src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -2384,45 +2386,49 @@ public Response getContentAccessBody(@Verify(Consumer.class) String consumerUuid
             throw new BadRequestException(i18n.tr("Content access mode does not allow this request."));
         }
 
-        SCACertificate cac = null;
+        SCACertificate x509Cert = scaCertificateGenerator.getX509Certificate(consumer);
+        ContentAccessPayload contentPayload = null;
         try {
-            cac = this.scaCertificateGenerator.generate(consumer);
+            contentPayload = scaCertificateGenerator.getContentPayload(consumer);
         }
         catch (ConcurrentContentPayloadCreationException e) {
-            // TODO: Handle as part of CHAINSAW-359
+            throw new ServiceUnavailableException(i18n.tr("Unable to create content access payload"), e);
```

Comment:
I thought the client is responding well for 429 status? Shouldn't we be throwing that instead of a 503?
Also, shouldn't we be setting the retry-after value?

Reply:
I forgot to add the retry-after header value. Updated to set the retry-after header value to 2 seconds. The design doc doesn't specify what the value should be, so if anyone want's a different time, I can change it.

For the status code, the design doc says ` This can be either a 429 or a 503`, and I went with a 503 because it is really a server error and not a client error.  However, I don't have a strong opinion on this. With the investigation of CHAINSAW-354 we know the client does not retry with a 429 response code.

Reply:
to be clear, the reason I mention 429 was beause I thought that even though the client wasn't retrying, it was at least handling it and printing out a user-friendly error (based on the investigation in CHAINSAW-354), while we don't know if 503 is being handled at all and may result in a stacktrace and even a potentially broken state in the client. If we prefer 503 we should at least do the same investigation for that status code first and see how the client reacts (for each of the 3 affected endpoints)